modules/couchbase/src/main/java/org/testcontainers/couchbase/CouchbaseContainer.java

```diff
@@ -353,6 +355,24 @@ private void createBuckets() {
                 .forStatusCode(200)
                 .waitUntilReady(this);
 
+            if (enabledServices.contains(CouchbaseService.QUERY)) {
+                // If the query service is enabled, make sure that we only proceed if the query engine also
+                // knows about the bucket in its metadata configuration.
+                Unreliables.retryUntilTrue(1, TimeUnit.MINUTES, () -> {
+                    Response queryResponse = doHttpRequest(QUERY_PORT, "/query/service", "POST", new FormBody.Builder()
```

Comment:
Just a thought: all these `Response` objects ought to be closed (here and elsewhere). Otherwise we'll potentially be holding connections open on the client and server side.

This is probably much more important inside a `retryUntilTrue` invocation, as the frequency of retries could be high.

Lombok's `@Cleanup` annotation would be a cheap and easy way to ensure that these response objects get cleaned up.

Reply:
@daschl ping

Reply:
sorry that I've been slow on this. Working on it now, will update the PR